src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+use Zookeeper;
+use Throwable;
+
+/**
+ * ZookeeperStore is a StoreInterface implementation using Zookeeper as store engine.
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ */
+class ZookeeperStore implements StoreInterface
+{
+    /**
+     * @var \Zookeeper
+     */
+    private $zookeeper;
+
+    /**
+     * @var string A prefix for the locking resource to identify the process
+     */
+    private $prefix;
+
+    /**
+     * A base node within which all the locks for PHP application are managed in the Zookeeper environment.
+     */
+    const LOCK_NODE = '/PHP_LOCK_NODE';
```

Comment:
Can you add a modifier? I believe this version of Symfony supports it.

Reply:
Made it private.